dronegen/windows.go

```diff
@@ -0,0 +1,226 @@
+	return step{
+		Name: "Check out Teleport",
+		Environment: map[string]value{
+			"WORKSPACE_DIR": {raw: workspace},
+		},
+		Commands: []string{
+			`$ErrorActionPreference = 'Stop'`,
+			`$Env:GOCACHE = "` + perBuildWorkspace + `/gocache"`,
+			`$TeleportSrc = "` + perBuildTeleportSrc + `"`,
+			`$TeleportRev = if ($Env:DRONE_TAG -ne $null) { $Env:DRONE_TAG } else { $Env:DRONE_COMMIT }`,
+			`New-Item -Path $TeleportSrc -ItemType Directory | Out-Null`,
+			`cd $TeleportSrc`,
+			`git clone https://github.com/gravitational/${DRONE_REPO_NAME}.git .`,
+			`git checkout $TeleportRev`,
+			`$WebappsSrc = "` + perBuildWebappsSrc + `"`,
+			`New-Item -Path $WebappsSrc -ItemType Directory | Out-Null`,
+			`cd $WebappsSrc`,
+			`git clone https://github.com/gravitational/webapps.git .`,
+			`git checkout $(go run $TeleportSrc/build.assets/tooling/cmd/get-webapps-version/main.go)`,
+		},
+	}
+}
+
+func updateWindowsSubreposStep(workspace string) step {
+	return step{
+		Name: "Checkout Submodules",
+		Environment: map[string]value{
+			"WORKSPACE_DIR":      {raw: workspace},
+			"GITHUB_PRIVATE_KEY": {fromSecret: "GITHUB_PRIVATE_KEY"},
+		},
+		Commands: []string{
+			`$Workspace = "` + perBuildWorkspace + `"`,
+			`$TeleportSrc = "` + perBuildTeleportSrc + `"`,
+			`. "$TeleportSrc/build.assets/windows/build.ps1"`,
+			`Enable-Git -Workspace $Workspace -PrivateKey $Env:GITHUB_PRIVATE_KEY`,
+			`cd $TeleportSrc`,
+			`git submodule update --init e`,
+			`git submodule update --init --recursive webassets`,
+			`Reset-Git -Workspace $Workspace`,
+		},
+	}
+}
+
+func installWindowsNodeToolchainStep(workspacePath string) step {
+	return step{
+		Name:        "Install Node Toolchain",
+		Environment: map[string]value{"WORKSPACE_DIR": {raw: workspacePath}},
+		Commands: []string{
+			`$ProgressPreference = 'SilentlyContinue'`,
+			`$ErrorActionPreference = 'Stop'`,
+			`$TeleportSrc = "` + perBuildTeleportSrc + `"`,
+			`. "$TeleportSrc/build.assets/windows/build.ps1"`,
+			// We can't use make, as there are too many posix dependencies in our makefile
+			// to abstract away right now, so instead of `$(make -C $TeleportSrc/build.assets print-node-version)`,
+			// we will just hardcode it for now
+			`$NodeVersion = "` + windowsNodeVersion + `"`,
+			`Install-Node -NodeVersion $NodeVersion -ToolchainDir "` + windowsToolchainDir + `"`,
+		},
+	}
+}
+
+func buildWindowsTeleportConnectStep(workspace string) step {
+	return step{
+		Name: "Build Teleport Connect",
+		Environment: map[string]value{
+			"WORKSPACE_DIR": {raw: workspace},
+		},
+		Commands: []string{
+			`$Workspace = "` + perBuildWorkspace + `"`,
+			`$TeleportSrc = "` + perBuildTeleportSrc + `"`,
+			`$WebappsSrc = "` + perBuildWebappsSrc + `"`,
+			`$NodeVersion = "` + windowsNodeVersion + `"`,
+			`. "$TeleportSrc/build.assets/windows/build.ps1"`,
+			`Enable-Node -NodeVersion $NodeVersion -ToolchainDir "` + windowsToolchainDir + `"`,
+			`cd $WebappsSrc`,
```

Comment:
Are we going to use `electron-builder` for code signing? It only requires us to set the [`CSC_LINK` variable](https://www.electron.build/code-signing).

Reply:
I'm looking in to it.

Reply:
Yes, I think it should even work without `file://` (`CSC_LINK=path-to-cert`). 
Another option is to encode it as base64 and keep it in the env variable, if that would be more convenient for us.

Reply:
I think the secret is already encoded in base64, so perhaps that's the way to do it